csrc/scheduler/greedy.cpp

```diff
@@ -342,7 +329,7 @@ class CompileTimeChecker : private IterVisitor {
     // not be parallelized with TID.
     auto out_tv = ir_utils::getTvOutput(scatter);
     checkDomainConstraints(
-        out_tv->domain()->initialLoop(), {constrained_out_logical_dim});
+        out_tv->domain()->logical(), {constrained_out_logical_dim});
```

Comment:
This should check the output logical domain. The loop domain is also necessary to check, but the index logical is also checked below, so that should be enough.